doc/source/serve/model_composition.md

```diff
@@ -72,8 +79,53 @@ Composition lets you break apart your application and independently scale each p
 With composition, you can avoid application-level bottlenecks when serving models and business logic steps that use different types and amounts of resources.
 :::
 
+### ServeHandle Deep Dive
+
+Conceptually, ServeHandle is a client side load balancer, routing requests to any replicas of a given deployment. Also, it performs buffering internally so it won't overwhelm the replicas.
+Using the current number of requests buffered, it informs the autoscaler to scale up the number of replicas.
+
+![architecture-diagram-of-serve-handle](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/serve-handle-explainer.png)
+
+ServeHandle takes request parameters and returns a future object of type [`ray.ObjectRef`](objects-in-ray), whose value will be filled with the result object. Because of the internal buffering, the time from submitting a request to getting a `ray.ObjectRef` varies from instantaneous to indefinitely long.
```

Comment:
> getting a `ray.ObjectRef` varies from instantaneous to indefinitely long.

This is true, but at first glance, it's a bit scary-sounding since it sounds like a request could hang indefinitely. Is there any way to rephrase here?

Reply:
yeah will do